[Tools] run-webkit-app command not able to set DYLD_FRAMEWORK_PATH under SIP-enabled environment.

megabitsenmzq opened the pull request:
#### 34e6542dbf34dbd110a8004e24b392b5da665b13
<pre>
[Tools] run-webkit-app command not able to set DYLD_FRAMEWORK_PATH under SIP-enabled environment.
<a href="https://bugs.webkit.org/show_bug.cgi?id=275207">https://bugs.webkit.org/show_bug.cgi?id=275207</a>

Reviewed by Alexey Proskuryakov.

While using the &quot;run-webkit-app&quot; command to run a demo app with self-built WebKit on macOS, the app is still using the system WebKit. The WebKit I built is not showing under &quot;Open Files and Ports&quot; in the  &quot;Activity Monitor&quot;. You can check this thread for more information: <a href="https://bugs.webkit.org/show_bug.cgi?id=274818">https://bugs.webkit.org/show_bug.cgi?id=274818</a>

After some hints from Alexey Proskuryakov, I fixed this bug by passing $ENV to the &quot;open&quot; command with some &quot;--env&quot; arguments. The &quot;open&quot; command couldn&apos;t keep environment variables while SIP is enabled.

* Tools/Scripts/webkitdirs.pm:
(commandLineArgumentsForRestrictedEnvironmentVariables):
(runMacWebKitApp):
(archCommandLineArgumentsForRestrictedEnvironmentVariables): Deleted.

[Tools] Remove old code and refactor.
<a href="https://bugs.webkit.org/show_bug.cgi?id=275207">https://bugs.webkit.org/show_bug.cgi?id=275207</a>

Reviewed by Alexey Proskuryakov.

Modified according to the reviewer&apos;s suggestion.

* Tools/Scripts/webkitdirs.pm:
(commandLineArgumentsForRestrictedEnvironmentVariables):

Canonical link: <a href="https://commits.webkit.org/279822@main">https://commits.webkit.org/279822@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d4a1792a3d7171daffc62743efaf27250ac88b8a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54644 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34066 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7219 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57924 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5377 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/56945 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41605 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5385 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44261 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3629 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56739 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32206 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47332 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25386 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29003 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4666 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3517 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/48019 "Built successfully and passed tests") | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4880 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59514 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/54150 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29882 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5024 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51685 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31027 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47418 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51067 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11964 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32014 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/66447 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30814 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12667 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Tools/Scripts/webkitdirs.pm:
```diff
@@ -3352,12 +3352,13 @@ sub runIOSWebKitApp($)
     die "Not using an iOS SDK."
 }
 
-sub archCommandLineArgumentsForRestrictedEnvironmentVariables()
```
This subroutine is exported at the top of the file, please remove the export. I verified that it's not needed any more.

```
       &archCommandLineArgumentsForRestrictedEnvironmentVariables
```

megabitsenmzq replied:
Do I need to add the new subroutine to the export, or just removing the old one is enough? 

(Btw, this is the first time I write Perl, so I might ask some silly questions😁.)

aproskuryakov replied:
Please just remove it - it was used in other files before, but not any more.

aproskuryakov commented on Tools/Scripts/webkitdirs.pm:
```diff
@@ -3352,12 +3352,13 @@ sub runIOSWebKitApp($)
-sub archCommandLineArgumentsForRestrictedEnvironmentVariables()
+sub commandLineArgumentsForRestrictedEnvironmentVariables($)
 {
+    my ($prefix) = @_;
```
We usually write this as `my $prefix = shift;`, but this is perfectly fine too.

aproskuryakov commented on Tools/Scripts/webkitdirs.pm:
```diff
@@ -3352,12 +3352,13 @@ sub runIOSWebKitApp($)
     foreach my $key (keys(%ENV)) {
         if ($key =~ /^DYLD_/) {
-            push @arguments, "-e", "$key=$ENV{$key}";
+            push @arguments, $prefix, "$key=$ENV{$key}";
```
This change is also perfectly fine, although I'd probably rewrite this with a combination of `grep` and `map`, just to show off :-)

aproskuryakov reviewed (changes requested):
Thank you for the PR! Please make the one change requested, and consider the other comments.

megabitsenmzq committed:
[Tools] run-webkit-app command not able to set DYLD_FRAMEWORK_PATH under SIP-enabled environment.
https://bugs.webkit.org/show_bug.cgi?id=275207

Reviewed by Alexey Proskuryakov.

While using the "run-webkit-app" command to run a demo app with self-built WebKit on macOS, the app is still using the system WebKit. The WebKit I built is not showing under "Open Files and Ports" in the  "Activity Monitor". You can check this thread for more information: https://bugs.webkit.org/show_bug.cgi?id=274818

After some hints from Alexey Proskuryakov, I fixed this bug by passing $ENV to the "open" command with some "--env" arguments. The "open" command couldn't keep environment variables while SIP is enabled.

* Tools/Scripts/webkitdirs.pm:
(commandLineArgumentsForRestrictedEnvironmentVariables):
(runMacWebKitApp):
(archCommandLineArgumentsForRestrictedEnvironmentVariables): Deleted.

[Tools] Remove old code and refactor.
https://bugs.webkit.org/show_bug.cgi?id=275207

Reviewed by Alexey Proskuryakov.

Modified according to the reviewer's suggestion.

* Tools/Scripts/webkitdirs.pm:
(commandLineArgumentsForRestrictedEnvironmentVariables):

Canonical link: https://commits.webkit.org/279822@main
